TST: Check float format in object column (#35603)

mdhsieh opened the pull request:
- [ ] closes #35603
- [x] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

This test checks if a Series cast to object still shows data in float format.

Ran `pytest pandas/tests/series/test_constructors.py`
Output:
```
================================================= test session starts =================================================
platform win32 -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: C:\Users\mdhsi\pandas-michael, configfile: pyproject.toml
plugins: hypothesis-6.12.0, asyncio-0.14.0, cov-2.11.1, forked-1.3.0, instafail-0.4.1, xdist-2.2.1
collected 301 items

pandas\tests\series\test_constructors.py .......................................................................................................................................................................................................x.....................................................................................................

--------------------------- generated xml file: C:\Users\mdhsi\pandas-michael\test-data.xml ---------------------------
================================================ slowest 30 durations =================================================
0.25s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_empty_constructor[<lambda>-True0]
0.13s call     pandas/tests/series/test_constructors.py::TestSeriesConstructorIndexCoercion::test_series_constructor_datetimelike_index_coercion
0.09s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['dateutil/US/Pacific'-True]
0.04s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_dtype_datetime64_10
0.02s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_empty[OrderedDict]
0.02s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_empty[dict]
0.02s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_cant_cast_datetimelike[PeriodIndex]
0.02s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_cant_cast_datetimelike[DatetimeIndex]
0.02s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_cant_cast_datetimelike[TimedeltaIndex]
0.02s setup    pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_datetimelike_scalar_to_string_dtype[string]
0.02s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_pass_none
0.02s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_categorical_with_coercion
0.02s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_dtype_timedelta64
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive[pytz.FixedOffset(300)-False]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['Asia/Tokyo'-True]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive[pytz.FixedOffset(300)-True]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive[tzlocal()-False]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['dateutil/Asia/Singapore'-False]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive[datetime.timezone(datetime.timedelta(seconds=3600))-False]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive[tzutc()-False]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['+01:15'-True]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['UTC'-False]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['UTC-02:15'-True]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive[tzlocal()-True]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive[pytz.FixedOffset(-300)-True]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['dateutil/Asia/Singapore'-True]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['UTC'-True]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive['-02:15'-False]
0.01s call     pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_data_aware_dtype_naive[pytz.FixedOffset(-300)-False]
=========================================== 300 passed, 1 xfailed in 2.22s ============================================
```

mdhsieh committed:
TST: Check Series when cast to object has float in object column (#35603).

mdhsieh committed:
TST: Assert equality with float was changed to assert string has decimal point.

May not be good idea to
use == to compare floating-point numbers.

pep8speaks commented:
Hello @mdhsieh! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/tests/series/test_constructors.py`](https://github.com/pandas-dev/pandas/blob/3f14a966e457ee06fe4f25eeed379390059efe5a/pandas/tests/series/test_constructors.py):

> [Line 1495:1](https://github.com/pandas-dev/pandas/blob/3f14a966e457ee06fe4f25eeed379390059efe5a/pandas/tests/series/test_constructors.py#L1495): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 1504:1](https://github.com/pandas-dev/pandas/blob/3f14a966e457ee06fe4f25eeed379390059efe5a/pandas/tests/series/test_constructors.py#L1504): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

mdhsieh committed:
TST: Remove whitespace before and after the new test.

To fix PEP8 issues.

mzeitlin11 commented:
Thanks for the pr @mdhsieh! If you look at the linked issue, we actually want to be testing output formatting, not construction. The test should essentially make sure that printing the series from the issue gives the expected result (hardcoded as a string).

mdhsieh commented:
Oh ok, let me change it. Should the test be moved to `tests/series/test_repr` folder then?

mdhsieh committed:
TST: Remove new test.

Needs to be rewritten to check formatting not construction.

mdhsieh committed:
TST: Rewrite and move new test to test_repr folder (#35603).

Checks if series representation matches expected String.

mzeitlin11 commented:
> Oh ok, I'll change it. Should the test be moved to `tests/series/test_repr` folder then?

Either there or `pandas/tests/io/formats/test_format.py`, not 100% sure so whichever looks like it has more similar tests to the one you write. (will be easy to change location once test is written, so don't worry too much about it :)

jreback commented on pandas/tests/series/test_repr.py:
```diff
@@ -240,6 +240,15 @@ def test_series_repr_nat(self):
         )
         assert result == expected
 
+    def test_float_repr_in_frame(self):
```
can you remove _in_frame (as this a series).

mdhsieh replied:
Ok, sure.

jreback reviewed (changes requested):
pls merge master and fix the linting issues.

mdhsieh committed:
TST: Remove in_frame from function name.

It is a Series not a DataFrame.

mdhsieh committed:
Merge remote-tracking branch 'upstream/master' into 35603-tst-series-object-float

mdhsieh committed:
tst: Remove parentheses to try fix linting issue.

They should not be there.

mdhsieh committed:
TST: Add newline to fix linting issue.

Failed pre-commit check since want 2 newlines before defining class,
but only had 1.

jreback commented:
thanks @mdhsieh
